### PR TITLE
[3577] [studio] /studio/api/2/monitoring API enhancement

### PR DESCRIPTION
**Pull request by dejan-brkic**
Added authorization token for monitoring APIs

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/3577

### CODE REVIEW

**Comment by sumerjabri** [comment]
Pending other token authorization work.

File: utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java [hunk 1]
```diff
@@ -36,18 +41,34 @@
...
 
     @GetMapping(ROOT_URL + MEMORY_URL)
-    public MemoryInfo getCurrentMemory() {
-        return MemoryInfo.getCurrentMemory();
+    public MemoryInfo getCurrentMemory(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return MemoryInfo.getCurrentMemory();
+        } else {
+            throw new InvalidMonitoringTokenException("Invalid token for monitoring authorization");
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java]: Change the message of the exception to `Monitoring authorization failed, invalid token`.

File: utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java [hunk 2]
```diff
@@ -36,18 +41,34 @@
...
 
     @GetMapping(ROOT_URL + STATUS_URL)
-    public StatusInfo getCurrentStatus() {
-        return StatusInfo.getCurrentStatus();
+    public StatusInfo getCurrentStatus(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return StatusInfo.getCurrentStatus();
+        } else {
+            throw new InvalidMonitoringTokenException("Invalid token for monitoring authorization");
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java]: Change the message of the exception to `Monitoring authorization failed, invalid token`.

File: utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java [hunk 3]
```diff
@@ -36,18 +41,34 @@
...
 
     @GetMapping(ROOT_URL + VERSION_URL)
-    public VersionInfo getCurrentVersion() throws Exception {
-        return VersionInfo.getVersion(this.getClass());
+    public VersionInfo getCurrentVersion(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException, IOException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return VersionInfo.getVersion(this.getClass());
+        } else {
+            throw new InvalidMonitoringTokenException("Invalid token for monitoring authorization");
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java]: Change the message of the exception to `Monitoring authorization failed, invalid token`.

File: utilities/src/main/java/org/craftercms/commons/exceptions/InvalidMonitoringTokenException.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.craftercms.commons.exceptions;
+
+/**
+ * Exception thrown when monitoring API is called with invalid authorization token
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/exceptions/InvalidMonitoringTokenException.java]: Change monitoring to management.

File: utilities/src/main/java/org/craftercms/commons/exceptions/InvalidMonitoringTokenException.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+ */
+public class InvalidMonitoringTokenException extends Exception {
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/exceptions/InvalidMonitoringTokenException.java]: Change monitoring to management.

File: utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java [hunk 6]
```diff
@@ -36,18 +41,34 @@
...
 
     @GetMapping(ROOT_URL + MEMORY_URL)
-    public MemoryInfo getCurrentMemory() {
-        return MemoryInfo.getCurrentMemory();
+    public MemoryInfo getCurrentMemory(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return MemoryInfo.getCurrentMemory();
+        } else {
+            throw new InvalidMonitoringTokenException("Monitoring authorization failed, invalid token.");
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java]: Change monitoring to management.

File: utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java [hunk 7]
```diff
@@ -36,18 +41,34 @@
...
 
     @GetMapping(ROOT_URL + STATUS_URL)
-    public StatusInfo getCurrentStatus() {
-        return StatusInfo.getCurrentStatus();
+    public StatusInfo getCurrentStatus(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return StatusInfo.getCurrentStatus();
+        } else {
+            throw new InvalidMonitoringTokenException("Monitoring authorization failed, invalid token.");
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java]: Change monitoring to management.

File: utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java [hunk 8]
```diff
@@ -36,18 +41,34 @@
...
 
     @GetMapping(ROOT_URL + VERSION_URL)
-    public VersionInfo getCurrentVersion() throws Exception {
-        return VersionInfo.getVersion(this.getClass());
+    public VersionInfo getCurrentVersion(@RequestParam(name = "token", required = true) String token)
+            throws InvalidMonitoringTokenException, IOException {
+        if (StringUtils.isNotEmpty(token) && StringUtils.equals(token, getConfiguredToken())) {
+            return VersionInfo.getVersion(this.getClass());
+        } else {
+            throw new InvalidMonitoringTokenException("Monitoring authorization failed, invalid token.");
```
Comment (avasquez614) [inline utilities/src/main/java/org/craftercms/commons/monitoring/rest/MonitoringRestControllerBase.java]: Change monitoring to management.

**Pull request merged**